<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>758</width>
    <height>378</height>
   </rect>
  </property>
  <property name="baseSize">
   <size>
    <width>688</width>
    <height>378</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Super Decoder</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icons/binary.png</normaloff>icons/binary.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">font: 8pt &quot;Segoe UI Historic&quot;;</string>
  </property>
  <property name="iconSize">
   <size>
    <width>32</width>
    <height>32</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="1" column="1">
     <widget class="QGroupBox" name="hash_options_group">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>185</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>185</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Hashing Algorithm Options&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="title">
       <string>Hash Options</string>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QCheckBox" name="crc32_check">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Regular CRC32&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>CRC32</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QCheckBox" name="sha256_check">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;SHA256 Hash&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>SHA256</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="adler_check">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Adler32 Hash&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Adler32</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="b64_256_check">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;SHA256 Hash Base64 Encoded&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>B64 SHA256</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="md5_check">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;MD5 Hash&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>MD5</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QCheckBox" name="sha1_check">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;SHA1 Hash&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>SHA1</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="options_group">
      <property name="minimumSize">
       <size>
        <width>185</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>185</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Options</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QComboBox" name="func_select">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select Function&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <item>
            <property name="text">
             <string>ASCII to Hex</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Hex to ASCII</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Decimal to Hex</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Decimal to Binary</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Base64 Encode</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Base64 Decode</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Reverse Nibble</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Switch Endianness</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ROT13</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Hash Text</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Find OUI Vendor</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Hex to Decimal IP</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="execute_btn">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Decode&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Execute</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="clear_btn">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Clear All&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0" rowspan="4">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="input_group">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter value to decode here&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="title">
         <string>Input Values</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLineEdit" name="input_line">
             <property name="minimumSize">
              <size>
               <width>250</width>
               <height>23</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>5</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="paste_btn">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Paste&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>icons/paste.png</normaloff>icons/paste.png</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="output_group">
        <property name="title">
         <string>Output Window</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0" rowspan="3">
            <widget class="QTextEdit" name="output_box">
             <property name="minimumSize">
              <size>
               <width>480</width>
               <height>0</height>
              </size>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Decoded output appears here&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>5</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="copy_btn">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Copy&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>icons/copy.png</normaloff>icons/copy.png</iconset>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="save_btn">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save to File&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>icons/save.png</normaloff>icons/save.png</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="1">
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="1">
     <widget class="QGroupBox" name="length_group">
      <property name="minimumSize">
       <size>
        <width>185</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>185</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Will either truncate excess, pad with appropriate character, or refuse to decode.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="title">
       <string>Improper Input Length Options</string>
      </property>
      <layout class="QFormLayout" name="formLayout_2">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QRadioButton" name="truncate_radio">
         <property name="text">
          <string>Truncate Incorrect Lengths</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="pad_radio">
         <property name="text">
          <string>Pad Incorrect Lengths</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QRadioButton" name="refuse_radio">
         <property name="text">
          <string>Refuse to Decode</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>758</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>input_line</tabstop>
  <tabstop>paste_btn</tabstop>
  <tabstop>output_box</tabstop>
  <tabstop>copy_btn</tabstop>
  <tabstop>save_btn</tabstop>
  <tabstop>func_select</tabstop>
  <tabstop>execute_btn</tabstop>
  <tabstop>clear_btn</tabstop>
  <tabstop>crc32_check</tabstop>
  <tabstop>adler_check</tabstop>
  <tabstop>md5_check</tabstop>
  <tabstop>sha1_check</tabstop>
  <tabstop>sha256_check</tabstop>
  <tabstop>b64_256_check</tabstop>
  <tabstop>truncate_radio</tabstop>
  <tabstop>pad_radio</tabstop>
  <tabstop>refuse_radio</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>input_line</sender>
   <signal>returnPressed()</signal>
   <receiver>execute_btn</receiver>
   <slot>animateClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>338</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>583</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>paste_btn</sender>
   <signal>clicked()</signal>
   <receiver>input_line</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>475</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>paste_btn</sender>
   <signal>clicked()</signal>
   <receiver>input_line</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>475</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copy_btn</sender>
   <signal>clicked()</signal>
   <receiver>output_box</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>475</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copy_btn</sender>
   <signal>clicked()</signal>
   <receiver>output_box</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>475</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_btn</sender>
   <signal>clicked()</signal>
   <receiver>input_line</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>666</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clear_btn</sender>
   <signal>clicked()</signal>
   <receiver>output_box</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>666</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>338</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
